### DISCUSSION TITLE
Docker: use own keys

### DISCUSSION THREAD

**Discussion by dsuesstr** [drakkan/sftpgo#588] [Q&A] [answered]
Hi @drakkan and Team,

Is it possible to use own pub/private keys for the server (not the users).
And if so, how/where do i need to map the keys to?

Can I map my own keys to /var/lib/sftpgo (id_ecdsa , id_ecdsa.pub, id_ed25519, id_ed25519.pub, id_rsa, id_rsa.pub) when running SFTPGO in a docker container?

**Comment by drakkan** [comment, marked as answer]
Hi,

yes this is supported, from the docs [here](https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md)

> host_keys, list of strings. It contains the daemon's private host keys. Each host key can be defined as a path relative to the configuration directory or an absolute one. If empty, the daemon will search or try to generate id_rsa, id_ecdsa and id_ed25519 keys inside the configuration directory. If you configure absolute paths to files named id_rsa, id_ecdsa and/or id_ed25519 then SFTPGo will try to generate these keys using the default settings.

So basically you can map your own host keys as you described above and it should work
